.github/workflows/pythonpackage.yml

```diff
@@ -43,9 +42,15 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements.txt
```

Comment:
It would be nice to have a requirements.txt that contains the deps needed for development e.g pytest, coveralls etc. and not have them internal to the particular CI.

Reply:
Yes, I added a requirements_dev.txt that contains all test/dev dependencies.

Reply:
Why not just call it requirements.txt? They should be used for development ref: https://stackoverflow.com/a/43659126 We should also enable `pip install -r requirements.txt` here and remove the individual pip installs in each step.

Reply:
I always used requirements.txt to pin or put constraints on the project dependencies. And other requirement files for dev/doc dependencies. Something like this (a sub-link from your so reference):
https://stackoverflow.com/questions/17803829/how-to-customize-a-requirements-txt-for-multiple-environments/20720019#20720019
But I am not sure what it the good way to do. I just the name to requirements.txt, that's fine for me.

> We should also enable pip install -r requirements.txt here and remove the individual pip installs in each step.

I already removed all "pip  install" and replaced them with one requirements install on each job. Do you mean something else ?

Reply:
Thanks. It look good now!